pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+	}
+
+	// 3. iterate through netnamespaces
+	//    - any with egressIP configured will cause an egressIP ovn resource to be created via k8s api
+	//    - a corresponding namespace label will be added to match the egressIP resource's namespace selector field
+	for i, nns := range nnsList.Items {
+		if netNamespaceHasEgressIpConfig(nns) {
+			// config detected, translating to ovn
+			// - delete cloudprivateipconfig if it exists
+			// - create egressip custom resource for ovn
+			if err := deleteCloudPrivateIpConfigs(ctx, client, nns.Object["egressIPs"].([]interface{})); err != nil {
+				return err
+			}
+
+			egressIpName := fmt.Sprint("egressip-", i)
+
+			egressIP := &uns.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "k8s.ovn.org/v1",
+					"kind":       "EgressIP",
+					"metadata": map[string]interface{}{
+						"name": egressIpName,
+					},
+					"spec": map[string]interface{}{
+						"egressIPs": nns.Object["egressIPs"].([]interface{}),
+						"namespaceSelector": map[string]interface{}{
+							"matchLabels": map[string]interface{}{
+								"kubernetes.io/metadata.name": nns.Object["netname"],
+							},
+						},
+					},
+				},
+			}
+			if err := apply.ApplyObject(ctx, client, egressIP, ""); err != nil {
+				return err
+			}
+
+			// now remove egressIP config from netnamespace
+			nns.Object["egressIPs"] = nil
+			if err := apply.ApplyObject(ctx, client, &nns, ""); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil // success
+}
+
+func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Client) error {
+	egressIpRollbackReady, err := sdnEgressIpResourcesReady(ctx, client)
+	if !egressIpRollbackReady {
+		return nil // wait for all SDN egressIP resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for egressips
+	eipList, err := client.Default().Dynamic().Resource(gvrEgressIp).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 2. query for nodes
+	nodeList, err := client.Default().Kubernetes().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 3. iterate through egressIP objects and generate netnamespaces
+	//    - value of `namespaceSelector` will determine netnamespace's name
+	//    - egressIP list will be copied to egressIPs field for netnamespace
+	for _, eip := range eipList.Items {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		if err := deleteCloudPrivateIpConfigs(ctx, client, egressIps); err != nil {
+			return err
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
```

Comment:
There are utility methods in `"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"` that might help having cleaner & more robust code when accessing or manipulating unstructured objects and is what is typically used for this.

Reply:
I'm not familiar with these libraries but that sounds way less painful. Leaving this comment unresolved for now since it's a lower priority compared to some other TODOs.

Reply:
I looked into this a little bit, and it seems like the code will only benefit stylistically from using said utility methods. That's definitely valuable, but I still think this is low priority for now given the 4.12 timeline - the code is overall still pretty readable (more imporantly, it is _functional_) and I want to focus more on e2e testing as the deadline approaches.